app/models/library_event.rb

```diff
@@ -0,0 +1,43 @@
+class LibraryEvent < ApplicationRecord
+  belongs_to :library_entry, required: true
+  belongs_to :user, required: true
+  belongs_to :anime
+  belongs_to :manga
+  belongs_to :drama
+
+  enum event: %i[added updated]
+  enum status: LibraryEntry.statuses
+  # changes should validate only on :update
+
+  # remove validation of changed_data
+  validates :event, presence: true
+
+  def self.create_for(event, library_entry)
+    # TODO: one edge case we might need to deal with
+    # is if someone changes progress up and down
+    # just to mess with it (other fields also)
+    LibraryEvent.create!(
+      notes: library_entry&.notes,
+      nsfw: library_entry.nsfw,
+      private: library_entry.private,
+      progress: library_entry.progress,
+      rating: library_entry&.rating,
+      reconsuming: library_entry.reconsuming,
+      reconsume_count: library_entry.reconsume_count,
+      volumes_owned: library_entry.volumes_owned,
+      time_spent: library_entry.time_spent,
+      status: library_entry.status,
+      # instead of polymorphic media
+      anime_id: library_entry&.anime_id,
+      manga_id: library_entry&.manga_id,
+      drama_id: library_entry&.drama_id,
```

Comment:
The question would be though, do we need access to anything isde those fields from JSONAPI, because removing those would kill the relationship it has.